### PR TITLE
Rebinning extinction curves

### LINKED ISSUE #87: Bug in rebinning function of extinction curves
I found a bug in the code to rebin extinction curves. The current code can be found here:
https://github.com/karllark/measure_extinction/blob/d88e8e8079601cb63fa97041dbac92bf222dc99a/measure_extinction/extdata.py#L23-L31

Because of the -1 in the index of a, you are not taking the mean of all the desired data.
It is easiest to see what happens if you rebin with a factor of 2. Let's say you want to calculate the 5th element of the rebinned array, i.e. i=5, rebin_fac =2.
Then `np.mean(a[i * rebin_fac : ((i + 1) * rebin_fac) - 1])` will calculate the mean of a[10:11]. However this is just 1 element, and not 2, because indexing of arrays is non-inclusive the last element: a[10:11] = a[10]. It is thus simply taking every second element in the array, instead of calculating the mean of 2 elements.
If you rebin with a factor of 3, it is taking the mean of 2 elements and ignoring the 3rd element, etc.

This can be solved by removing the -1, or by using a different way to rebin an array, e.g. with scipy.stats.binned_statistic(). Both methods give the same result for a rebinning factor of 2. If you go to higher rebinning factors, the result is not exactly the same, because of the way the elements are collected into bins. I would argue that the scipy method is the more accurate one, because of the way it deals with bin edges.

### PR DESCRIPTION
**Pull request by mdecleir**
I changed the way the extinction curves are rebinned.

The original code:
- had a bug related to slicing, see the discussion in issue #87.
- was calculating the mean of the same number of neighboring data points for every "bin". For non-equally spaced wavelength grids, one could thus potentially be taking the mean of values that are very far away.
- returned a nan if any of the values in the bin was nan.

The new code:
- uses a number of equally wide bins and calculates the mean wavelength and mean extinction in every bin. A different number of data points will be used in every bin if the wavelength grid is not equally spaced. This ensure that in sparse regions, not too much binning is happening.
- bins the wavelengths and extinction values at the same time.
- skips nan extinction values and the corresponding wavelengths, so that the same data points are used to calculate the mean wavelength and the mean extinction in a certain bin.

This PR closes #87.

### CODE REVIEW

**Comment by mdecleir** [comment]
This PR also contains the commits from PR #90, because I built it off from that branch. Not sure what will happen when you try to merge this one after the other one is merged.

**Comment by karllark** [comment]
Sound be fine.  The "extra" commits just won't happen.

File: measure_extinction/extdata.py
Comment (karllark) [inline measure_extinction/extdata.py:34]: Does this need to be extinction values?  Or could it also be for spectra?  Just wondering as would be useful to have a rebin for both.  No changes needed, just an idea that popped into my head when looking at the very nice changes.

Comment (mdecleir) [reply measure_extinction/extdata.py:34]: This could work for anything basically. I am just calling it extinction here because it is located in the extdata.py script. We could have a similar function in the stardata.py e.g.

**Review by karllark — APPROVED** [review]
Looks good to me.

**Pull request merged**
